Source/Core/Core/IOS/Device.cpp

```diff
@@ -77,6 +77,15 @@ IOCtlVRequest::IOCtlVRequest(const u32 address_) : Request(address_)
   }
 }
 
+const IOCtlVRequest::IOVector* IOCtlVRequest::GetVector(size_t index) const
+{
+  if (index >= (in_vectors.size() + io_vectors.size()))
+    return nullptr;
```

Comment:
I'm just wondering... could a custom IOCTLV request make ```GetVector()``` return ```nullptr``` in case where something is expected at a given index? If so, ```GetVector()``` calls should be checked if you don't want Dolphin to crash. Wouldn't it be better to be more verbose than just silently returning nullptr (an error log maybe) since it isn't supposed to happen I suppose?

Reply:
Since you use `GetVector(n)->size` etc. all over the place, that might not be a bad idea. Or perhaps, return an `std::optional` instead so you can use `value_or` (if there's a useful default you could fall back to?)
Or simply `assert` because it's not supposed to happen ;)

Reply:
Hmm, true, right now GetVector() is only called after VEN/HID has already checked the number of vectors, so the index being out-of-bounds is not supposed to happen.
I'll add an assert.